prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -250,19 +253,28 @@ class Hl7Serializer(val metadata: Metadata) : Logging {
                 var value = ""
                 for (i in 0 until hl7Fields.size) {
                     val hl7Field = hl7Fields[i]
-                    value = when {
+                    value =
                         // Decode a phone number
-                        element.type == Element.Type.TELEPHONE -> decodeHl7PhoneNumber(terser, element, hl7Field)
+                        if (element.type == Element.Type.TELEPHONE) {
+                            decodeHl7PhoneNumber(terser, element, hl7Field)
+                        }
                         // Decode a timestamp
-                        element.type == Element.Type.DATETIME -> decodeHl7DateTime(terser, element, hl7Field, warnings)
+                        else if (element.type == Element.Type.DATETIME ||
+                            element.type == Element.Type.DATE
+                        ) {
+                            decodeHl7DateTime(terser, element, hl7Field, warnings)
+                        }
                         // Decode an AOE question
-                        hl7Field == "AOE" -> decodeAOEQuestion(element, terser, errors, warnings)
+                        else if (hl7Field == "AOE") {
+                            decodeAOEQuestion(element, terser, errors, warnings)
+                        }
                         // No special case here, so get a value from an HL7 field
-                        else -> queryTerserForValue(
-                            terser, getTerserSpec(hl7Field), element.name,
-                            errors, warnings
-                        )
-                    }
+                        else {
+                            queryTerserForValue(
+                                terser, getTerserSpec(hl7Field), element.name,
+                                errors, warnings
+                            )
+                        }
                     if (value.isNotBlank()) break
```

Comment:
Not a fail, but I'm wondering why replace the when with an if/else. Just preference?

Reply:
I struggled with this one.  I researched online what to do when you have two when blocks with the same code.  The consensus online was to use if/else instead for readability.  No other reason.